src/toggle-switch/toggle-switch.scss

```diff
@@ -2,194 +2,206 @@
 // See LICENSE.md in the project root for license terms and full copyright notice.
 @import '../style/index';
 
+$toggle-switch-height: $iui-baseline * 2;
+
 @mixin iui-toggle-switch {
   @include iui-reset;
-  display: flex;
+  display: grid;
+  grid-template: 'toggle';
+  grid-column-gap: $iui-s;
   align-items: center;
   font-size: $iui-font-size;
   width: fit-content;
-  padding: ($iui-baseline/2) 0;
+  padding: ($iui-baseline * 0.5) 0;
   user-select: none;
   cursor: pointer;
   position: relative;
+  isolation: isolate;
   @include themed {
     color: rgba(t(iui-color-foreground-body-rgb), t(iui-opacity-2));
   }
 
-  > input {
-    width: 0;
-    height: 0;
+  > input[type='checkbox'] {
+    @include iui-reset;
+    @include iui-focus($thickness: 1px, $offset: 1px);
+    grid-area: toggle;
+    display: inline-block;
+    flex-shrink: 0;
+    position: relative;
     appearance: none;
-    opacity: 0;
-    position: absolute;
+    width: $toggle-switch-height * 2 - $iui-xs;
+    height: $toggle-switch-height;
+    border-radius: $toggle-switch-height * 0.5;
+    cursor: pointer;
+    @include themed {
+      background-color: t(iui-color-background-1);
+      border-color: rgba(t(iui-color-foreground-body-rgb), t(iui-opacity-4));
+    }
 
-    // Unchecked toggle switch
-    ~ .iui-toggle {
-      display: block;
-      position: relative;
-      height: $iui-baseline * 2 - $iui-xxs;
-      width: ($iui-m * 2) + ($iui-xxs * 3);
-      border-radius: $iui-baseline;
+    &::before {
+      content: '';
+      position: absolute;
+      inset: 0;
+      transition: border-color $iui-speed-fast ease-out;
+      border-radius: inherit;
+      border-style: solid;
+      border-width: 1px;
+      border-color: inherit;
+      box-sizing: border-box;
+    }
+
+    &::after {
+      content: '';
+      position: absolute;
+      height: $toggle-switch-height - ($iui-xxs * 3);
+      width: $toggle-switch-height - ($iui-xxs * 3);
+      aspect-ratio: 1 / 1;
+      top: 50%;
+      transform: translateY(-50%);
+      border-radius: 50%;
       @media (prefers-reduced-motion: no-preference) {
```

Comment:
background and opacity transitions dont need to be inside this media query

Reply:
`@media (prefers-reduced-motion: no-preference)` is only targeting motion now.